src/ucp/proto/proto_am.inl

```diff
@@ -258,6 +258,31 @@ ucs_status_t ucp_am_zcopy_common(ucp_request_t *req, const void *hdr,
                            &req->send.state.uct_comp);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_zcopy_single_handle_status(ucp_request_t *req,
+                               const ucp_dt_state_t *dt_state,
+                               ucs_status_t status,
+                               ucp_req_complete_func_t complete_ok)
+{
+    if (status == UCS_INPROGRESS) {
+        ucp_request_send_state_advance(req, dt_state,
+                                       UCP_REQUEST_SEND_PROTO_ZCOPY_AM,
+                                       UCS_INPROGRESS);
+        return UCS_OK;
+    }
+
+    if (status == UCS_OK) {
+        complete_ok(req, UCS_OK);
+        return UCS_OK;
+    }
+
+    ucs_assert(UCS_STATUS_IS_ERR(status));
+    ucp_request_send_state_advance(req, dt_state,
+                                   UCP_REQUEST_SEND_PROTO_ZCOPY_AM,
+                                   status);
+    return status;
```

Comment:
should return OK to remote from pending queue, except for NO_RESOURCE
also what about UCP_STATUS_PENDING_SWITCH?